src/Components/ABDM/LinkAbhaNumber/CreateWithAadhaar.tsx

```diff
@@ -0,0 +1,618 @@
+          onClick={handleSubmit}
+        >
+          {t("verify_otp")}
+        </ButtonV2>
+      </div>
+    </div>
+  );
+}
+
+type ILinkMobileNumberProps = InjectedStepProps<Memory>;
+
+function LinkMobileNumber({ memory, setMemory, next }: ILinkMobileNumberProps) {
+  const { t } = useTranslation();
+
+  const handleSubmit = async () => {
+    setMemory((prev) => ({ ...prev, isLoading: true }));
+
+    const { res, data, error } = await request(
+      routes.abdm.healthId.abhaCreateLinkMobileNumber,
+      {
+        body: {
+          mobile: memory?.mobileNumber.replace("+91", "").replace(/ /g, ""),
+          transaction_id: memory?.transactionId,
+        },
+      },
+    );
+
+    if (res?.status === 200 && data) {
+      setMemory((prev) => ({
+        ...prev,
+        transactionId: data.transaction_id,
+      }));
+      Notify.Success({
+        msg: data.detail ?? t("mobile_otp_send_success"),
+      });
+      next();
+    } else {
+      Notify.Error({
+        msg: data?.detail ?? error?.message ?? t("mobile_otp_send_error"),
+      });
+    }
+
+    setMemory((prev) => ({ ...prev, isLoading: false }));
+  };
+
+  return (
+    <div>
+      <div className="mt-0">
+        <label
+          htmlFor="mobile_number"
+          className="block text-base font-normal text-secondary-900"
+        >
+          {t("enter_mobile_number")}
+        </label>
+        <PhoneNumberFormField
+          labelSuffix={<></>}
+          name="mobile_number"
+          value={memory?.mobileNumber}
+          disabled={true}
+          onChange={() => null}
+          types={["mobile"]}
+        />
+      </div>
+
+      <p className="mb-4 text-sm text-secondary-800">
+        {t("mobile_number_different_from_aadhaar_mobile_number")}
+      </p>
+
+      <div className="mt-4 flex items-center">
+        <ButtonV2
+          className="w-full"
+          loading={memory?.isLoading}
+          onClick={handleSubmit}
+        >
+          {t("send_otp")}
+        </ButtonV2>
+      </div>
+    </div>
+  );
+}
+
+type IVerifyMobileNumberProps = InjectedStepProps<Memory>;
+
+function VerifyMobileNumber({
+  memory,
+  setMemory,
+  next,
+}: IVerifyMobileNumberProps) {
+  const { t } = useTranslation();
+  const [otp, setOtp] = useState("");
+
+  const handleSubmit = async () => {
+    setMemory((prev) => ({ ...prev, isLoading: true }));
+
+    const { res, data, error } = await request(
+      routes.abdm.healthId.abhaCreateVerifyMobileNumber,
+      {
+        body: {
+          transaction_id: memory?.transactionId,
+          otp: otp,
+        },
+      },
+    );
+
+    if (res?.status === 200 && data) {
+      setMemory((prev) => ({
+        ...prev,
+        transactionId: data.transaction_id,
+      }));
+      Notify.Success({
+        msg: data.detail ?? t("mobile_otp_verify_success"),
+      });
+      next();
+    } else {
+      Notify.Error({
+        msg: data?.detail ?? error?.message ?? t("mobile_otp_verify_error"),
+      });
```

Comment:
for 4xx/5xx errors, `data` would always be `undefined`.
```suggestion
      Notify.Error({
        msg: error?.message ?? t("mobile_otp_verify_error"),
      });
```